apps/frontend/src/libs/components/number-input/number-input.tsx

```diff
@@ -0,0 +1,73 @@
+import { IconButton } from "~/libs/components/components.js";
+import { useCallback } from "~/libs/hooks/hooks.js";
+
+import styles from "./styles.module.css";
+
+type Properties = {
+	onChange: (value: number) => void;
+	range: {
+		max: number;
+		min: number;
+	};
+	value: number;
+};
+
+const STEP_VALUE = 1;
+
+const NumberInput = ({
+	onChange,
+	range,
+	value,
+}: Properties): React.JSX.Element => {
+	const handleIncrement = useCallback((): void => {
+		const newValue = value + STEP_VALUE;
+		onChange(newValue);
+	}, [onChange, value]);
+
+	const handleDecrement = useCallback((): void => {
+		const newValue = value - STEP_VALUE;
+		onChange(newValue);
+	}, [onChange, value]);
+
+	const handleInputChange = useCallback(
+		(event: React.ChangeEvent<HTMLInputElement>) => {
+			const inputValue = Number(event.target.value);
+
+			onChange(inputValue);
+		},
+		[onChange],
+	);
+
+	return (
+		<div className={styles["container"]}>
+			<div className={styles["wrapper"]}>
+				<input
+					className={styles["input"]}
+					onChange={handleInputChange}
+					type="number"
+					value={value}
+				/>
+				<div className={styles["buttons"]}>
+					<IconButton
+						className={styles["button"]}
+						disabled={value >= range.max}
+						hiddenLabel="Increase value"
```

Comment:
Let's name it's just a label. For visible labels, we will use buttons and add an icon to them
```suggestion
						label="Increase value"
```